src/ipc/handler.c

```diff
@@ -918,6 +918,7 @@ static int ipc_probe_info(uint32_t header)
 	/* write data to the outbox */
 	if (params->rhdr.hdr.size <= MAILBOX_HOSTBOX_SIZE &&
 	    params->rhdr.hdr.size <= SOF_IPC_MSG_MAX_SIZE) {
+		params->rhdr.error = ret;
```

Comment:
Should be two patches, one for each fix (since they fix different things).

Reply:
done